app/src/main/java/ch/epfl/cs311/wanderwave/viewmodel/MapViewModel.kt

```diff
@@ -0,0 +1,33 @@
+package ch.epfl.cs311.wanderwave.viewmodel
+
+import androidx.lifecycle.ViewModel
+import ch.epfl.cs311.wanderwave.model.data.Beacon
+import ch.epfl.cs311.wanderwave.model.repository.BeaconRepositoryImpl
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+class MapViewModel @Inject constructor(private val repository: BeaconRepositoryImpl) : ViewModel() {
```

Comment:
I think a MapViewModel class is necessary, because I needed it too for the Map UI :+1: